### PR TITLE
Provides boilerplate simplification for SeBootstrap API

### PR DESCRIPTION
**Pull request by spericas**
Provides boilerplate simplification for SeBootstrap API:

1. New methods for default config
2. New methods for class instances
3. New method for shutdown hooks
4. New methods to obtain baseUri's
5. Removal of redundant public/final/etc as flagged by modern IDE's
6. Update to examples

See issue #1009 for an example.

### CODE REVIEW

**Review by mkarg — DISMISSED** [review]
Please provide tests for all new API methods. => -1

Please split the single commit into (at least) two commits as the changes are semantically *unrelated*:
* New API methods should go into one PR.
* IDE/Formatting changes should go into a second PR.
I would suggest to even split it into two PRs so we can put the formatting changes on the fast lane.

Side note: I doubt the usefulness of the new API `start` methods as least people would call them such often that it makes actual sense to have all of them. As JAX-RS is a foundational API only, least users will have a *measurable* benefit.

File: jaxrs-api/src/main/java/jakarta/ws/rs/SeBootstrap.java [hunk 1]
```diff
@@ -165,7 +232,7 @@
          *
          * @since 3.1
          */
-        static final String PROTOCOL = "jakarta.ws.rs.SeBootstrap.Protocol";
+        String PROTOCOL = "jakarta.ws.rs.SeBootstrap.Protocol";
```
[the remark this answers is not in the document]
Comment (mkarg) [reply jaxrs-api/src/main/java/jakarta/ws/rs/SeBootstrap.java:235]: Modern IDEs can also be configured to not underline it. If you fix it *here* there you need to fix it *everywhere*, so maybe it is best to provide a separate PR.

File: jaxrs-api/src/main/java/jakarta/ws/rs/SeBootstrap.java [hunk 2]
```diff
@@ -140,6 +143,70 @@
...
+     * environments is undefined.
+     * </p>
+     *
+     * @param application The application to start up.
+     * @return {@code CompletionStage} (possibly asynchronously) producing handle of the running application
+     * {@link SeBootstrap.Instance instance}.
+     * @see Configuration
+     * @since 3.1
+     */
+    static CompletionStage<Instance> start(final Application application) {
```
[the remark this answers is not in the document]
Comment (mkarg) [reply jaxrs-api/src/main/java/jakarta/ws/rs/SeBootstrap.java:160]: No, this PR is just the *place* where the discussion could run; "The" discussion is the sum of all answers. ;-)

What I ask for is that more committers review this and share their opinions.

[the remark this answers is not in the document]
Comment (spericas) [reply jaxrs-api/src/main/java/jakarta/ws/rs/SeBootstrap.java:161]: Yes, we need more people to review this.

File: jaxrs-api/src/main/java/jakarta/ws/rs/SeBootstrap.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (spericas) [reply jaxrs-api/src/main/java/jakarta/ws/rs/SeBootstrap.java:236]: I will not create another PR for this for the reasons mentioned above

File: jaxrs-api/src/main/java/jakarta/ws/rs/core/EntityPart.java [hunk 3]
```diff
@@ -129,8 +129,6 @@ static Builder withFileName(String partAndFileName) {
      * an {@code IllegalStateException} if it is called after calling
      * {@link #getContent} or {@link #getContent(GenericType)}.
      * 
-     * @param <T> type parameter of the value returned
```
[the remark this answers is not in the document]
Comment (spericas) [reply jaxrs-api/src/main/java/jakarta/ws/rs/core/EntityPart.java]: Should be restored

**Comment by spericas** [comment]
@mkarg If we agree on including these boilerplate simplifications, I will add more tests as you requested before finalizing the PR.

**Review by andymc12 — APPROVED** [review]
I think the changes are fine as-is.  I agree with @mkarg that we should have TCK tests for these new methods, but we haven't required them for previous API changes.  (That said, it might be good to add that as a requirement in the future once we have the TCK infrastructure in this repo.)

**Review by mkarg — COMMENTED** [review]
> @mkarg If we agree on including these boilerplate simplifications, I will add more tests as you requested before finalizing the PR.

Yes, we never had such a rule so I reduce my -1 to -0.

Now that we have a SeBootstrap TCK (and hopefully soon even a full TCK) it makes no sense to further add API changes *without* adding tests for them in the same PR. This keeps together the API change with the TCK change very nicely and if needed, both can be easily cherry-picked or undone without searching git history.

Your proposal sounds good to me, so basically I am not -1 but -0 once you squash your recent commits before mering and +0 once you add brief tests.

If you even separate the IDE changes and API changes into two PRs I would be even +1 for the API changes. I understand and respect your opinion, but there is *nothing* non-negotiable in Jakarta projects besides the rules given by the EF as all other rules are defined by a majority of committers. If we want to change the code style, we should discuss this separately. It is a no-go *for me* to mix API changes with style changes in the same *commit* actually as it makes undoing or cherry-picking harder or even conflicting.

**Comment by mkarg** [comment]
> (That said, it might be good to add that as a requirement in the future once we have the TCK infrastructure in this repo.)

Actually I *only* asked for the tests because *we already have* the TCK infrastracture *for SeBootstrap* in this repo **already**, so there is no good reason anymore to provide SeBootstrap-related API changes *without* tests.

**Review by mkarg — COMMENTED** [review]
LGTM! Thanks for the additional tests. As I said, I am +0 now, so I do not stand in the way of this PR if another committer actively is +1 for it (I personally will definitively be +1 as soon as the formatting changes go in a separate PR).

**Review by mkarg — COMMENTED** [review]
LGTM! Thanks for the additional tests. As I said, I am +0 now, so I do not stand in the way of this PR if another committer actively is +1 for it (I personally will definitively be +1 as soon as the formatting changes go in a separate PR).

**Review by andymc12 — DISMISSED** [review]
LGTM - thanks @spericas!

File: jaxrs-api/src/main/java/jakarta/ws/rs/SeBootstrap.java [hunk 4]
```diff
@@ -140,6 +144,70 @@
...
+     * @param clazz The application class.
+     * @param configuration Provides information needed for bootstrapping the application.
+     * @return {@code CompletionStage} (possibly asynchronously) producing handle of the running application
+     * {@link SeBootstrap.Instance instance}.
+     * @see Configuration
+     * @since 3.1
+     */
+    static CompletionStage<Instance> start(final Class<? extends Application> clazz, final Configuration configuration) {
+        try {
+            Application application = clazz.getDeclaredConstructor().newInstance();
```
Comment (jansupol) [inline jaxrs-api/src/main/java/jakarta/ws/rs/SeBootstrap.java:184]: Instantiation in a static method cannot be overridden by an implementation.  I assume the instantiation should better be handled there, possibly by the DI. Would it be better to add an extra method `RuntimeDelegate#bootstrap(final Class<? extends Application> application, final SeBootstrap.Configuration configuration)`?

Comment (spericas) [reply jaxrs-api/src/main/java/jakarta/ws/rs/SeBootstrap.java:184]: Yes, I'm glad you brought this up. I was unsure about that instantiation. Will update the PR.

**Review by mkarg — COMMENTED** [review]
New method RuntimeDelegate.bootstrap LGTM, and thanks a lot for including the unit test already! :-)

**Review by jansupol — APPROVED** [review]
LGTM, thank you

**Review by andymc12 — APPROVED** [review]
LGTM

**Pull request merged**